packages/snap-preact-components/src/components/Atoms/LoadMore/LoadMore.tsx

```diff
@@ -0,0 +1,299 @@
+									width: `${progressIndicatorWidth}px`,
+									height: `${progressIndicatorWidth}px`,
+									position: 'absolute',
+									borderRadius: '50%',
+								},
+								'& .ss__loadMore__progress__indicator__radial__circle__mask': {
+									clip: `rect(0px, ${progressIndicatorWidth}px, ${progressIndicatorWidth}px, ${progressIndicatorWidth! / 2}px)`,
+								},
+								'& .ss__loadMore__progress__indicator__radial__circle__mask__fill': {
+									transform: `rotate(${radialAngle}deg)`,
+								},
+							},
+							'& .ss__loadMore__progress__text': {
+								width: `${progressIndicatorWidth! - progressIndicatorSize!}px`,
+								height: `${progressIndicatorWidth! - progressIndicatorSize!}px`,
+								borderRadius: '50%',
+								background: '#fff',
+								lineHeight: `${progressIndicatorWidth! - progressIndicatorSize!}px`,
+								textAlign: 'center',
+								marginTop: `${progressIndicatorSize! / 2}px`,
+								marginLeft: `${progressIndicatorSize! / 2}px`,
+								position: 'absolute',
+								fontSize: `calc(${progressIndicatorWidth}px / ${Math.max(1, `${pagination!.end}`.length + `${pagination!.totalResults}`.length)})`,
+							},
+							'& .ss__loadMore__progress__indicator__radial__circle__mask': {
+								'& .ss__loadMore__progress__indicator__radial__circle__mask__fill': {
+									clip: `rect(0px, ${progressIndicatorWidth! / 2 + 0.5}px, ${progressIndicatorWidth}px, 0px)`,
+									backgroundColor: color || theme?.colors?.primary || '#ccc',
+								},
+								'&.ss__loadMore__progress__indicator__radial__circle__mask--full': {
+									transform: `rotate(${radialAngle}deg)`,
+								},
+							},
+						},
+					},
+				},
+			},
+		});
+	},
+};
+
+export const LoadMore = observer((properties: LoadMoreProps): JSX.Element => {
+	const globalTheme: Theme = useTheme();
+	const defaultProps: Partial<LoadMoreProps> = {
+		loadMoreText: 'Load More',
+		progressIndicator: 'bar',
+		progressIndicatorWidth: properties?.progressIndicator === 'radial' ? 70 : 300,
+		progressIndicatorSize: properties?.progressIndicator === 'radial' ? 10 : 5,
+	};
+
+	const props = mergeProps('loadMore', globalTheme, defaultProps, properties);
+
+	const {
+		pagination,
+		controller,
+		onClick,
+		auto,
+		intersectionOffset,
+		loading,
+		loadMoreText,
+		progressIndicator,
+		hideProgressIndicator,
+		hideProgressText,
+		disableStyles,
+		className,
+		style,
+		styleScript,
+	} = props;
+
+	const subProps: LoadMoreSubProps = {
+		button: {
+			// default props
+			className: 'ss__loadMore__button',
+			// global theme
+			...globalTheme?.components?.button,
+			// inherited props
+			...defined({
+				disableStyles,
+			}),
+			// component theme overrides
+			theme: props?.theme,
+		},
+	};
+
+	const store = pagination || controller?.store?.pagination;
+	const isLoading = typeof loading == 'boolean' ? loading : controller?.store?.loading;
+
+	if (!store) {
+		return <Fragment></Fragment>;
```

Comment:
could we just add this check to the main ternary on the return on line 200 and remove this? Since we return an empty fragment there already.

Reply:
This is here because the styles are using the store data